src/config/config.ts

```diff
@@ -106,83 +108,97 @@ export class Config extends ConfigFile<ConfigFile.Options> {
    * allows users to override the 10,000 result query limit
    */
   public static readonly MAX_QUERY_LIMIT = 'maxQueryLimit';
-  private static allowedProperties: ConfigPropertyMeta[];
-  private static messages: Messages;
-  private static propertyConfigMap: Dictionary<ConfigPropertyMeta>;
-  private crypto?: Crypto;
 
-  public constructor(options?: ConfigFile.Options) {
-    super(options || Config.getDefaultOptions(false));
-
-    if (!Config.messages) {
-      Config.messages = Messages.loadMessages('@salesforce/core', 'config');
-    }
+  private static get propertyConfigMap(): Dictionary<ConfigPropertyMeta> {
+    return keyBy(Config.allowedProperties, 'key');
+  }
 
-    if (!Config.allowedProperties) {
-      Config.allowedProperties = [
-        {
-          key: 'instanceUrl',
-          input: {
-            // If a value is provided validate it otherwise no value is unset.
-            validator: (value) => value == null || (isString(value) && sfdc.isSalesforceDomain(value)),
-            failedMessage: Config.messages.getMessage('InvalidInstanceUrl'),
-          },
+  private static allowedProperties: ConfigPropertyMeta[] = [
+    {
+      key: 'instanceUrl',
+      input: {
+        // If a value is provided validate it otherwise no value is unset.
+        validator: (value) => value == null || (isString(value) && sfdc.isSalesforceDomain(value)),
+        get failedMessage() {
```

Comment:
Out of curiosity, any reason you changed them to property getters?

Reply:
https://github.com/forcedotcom/sfdx-core/pull/320/files#diff-615dacd7000fa06d7afef0f22b7b6d9267c6219fbfc9fa0315e51a5f5d963755R198-R200

Making them lazy allows them to pick up `Config.messages` being set in the constructor.